src/librustc_feature/active.rs

```diff
@@ -559,6 +559,9 @@ declare_features! (
     /// Allow negative trait implementations.
     (active, negative_impls, "1.44.0", Some(68318), None),
 
+    /// Lazily evaluate constants. Which allows constants to depend on type parameters.
+    (active, lazy_normalization_consts, "1.44.0", Some(60471), None),
```

Comment:
I tested both `libcore` and `liballoc` with this feature which seemed to work. Should I go ahead and always enable lazy normalization when using the const generics feature?

This feature was added before the the const generics hack was intruced, so gating on `#![feature(const_generics)]` may be enough. https://github.com/rust-lang/rust/blob/63d03779946a07d8c418a361278168a6d3a1f4d2/src/librustc_typeck/collect.rs#L1175

Reply:
I think we probably want to enable lazy normalisation whenever we enable const generics. cc @eddyb

Reply:
The const generics feature is already a "warning, this doesn't work that well yet" sort of feature, right?

Reply:
That's right.